Helm/opensearch-dashboards/templates/deployment.yaml

```diff
@@ -0,0 +1,156 @@
+        - name: config
+          configMap:
+            name: {{ template "opensearch-dashboards.fullname" . }}-config
+        {{- end }}
+    {{- with .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+      tolerations:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.imagePullSecrets | indent 8 }}
+    {{- end }}
+      {{- if .Values.extraInitContainers }}
+      # Currently some extra blocks accept strings
+      # to continue with backwards compatibility this is being kept
+      # whilst also allowing for yaml to be specified too.
+      initContainers:
+      {{- if eq "string" (printf "%T" .Values.extraInitContainers) }}
+{{ tpl .Values.extraInitContainers . | indent 6 }}
+      {{- else }}
+{{ toYaml .Values.extraInitContainers | indent 6 }}
+      {{- end }}
+      {{- end }}
+      containers:
+      - name: dashboards
+        securityContext:
+{{ toYaml .Values.securityContext | indent 10 }}
+        image: "{{ .Values.image }}:{{ .Values.imageTag | default .Chart.AppVersion }}"
+        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        env:
+        {{- if .Values.opensearchURL }}
+          - name: OPENSEARCH_URL
+            value: "{{ .Values.opensearchURL }}"
+        {{- else if .Values.opensearchHosts }}
+          - name: OPENSEARCH_HOSTS
+            value: "{{ .Values.opensearchHosts }}"
+        {{- end }}
+          - name: SERVER_HOST
+            value: "{{ .Values.serverHost }}"
+            {{- if .Values.opensearchAccount.secret }}
+            - name: OPENSEARCH_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: username
+            - name: OPENSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: password
+            {{- if and .Values.opensearchAccount.keyPassphrase.enabled }}
+            - name: KEY_PASSPHRASE
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: keypassphrase
+            # 32-character random string to be used as cookie password by security plugin
+            {{- end }}
+            - name: COOKIE_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: cookie
+            {{- end }}
+{{- if .Values.extraEnvs }}
+{{ toYaml .Values.extraEnvs | indent 10 }}
+{{- end }}
+{{- if .Values.envFrom }}
+        envFrom:
+{{ toYaml .Values.envFrom | indent 10 }}
+{{- end }}
+        ports:
+        - containerPort: {{ .Values.service.port }}
```

Comment:
We should give this a name that can be referenced by the service. More human readable.

Reply:
Did you mean to reference this as `targetPort`?

Reply:
@TheAlgo https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#containerport-v1-core

```sh
containerPort: {{ .Values.service.port }}
name: http
protocol: TCP
```

Reply:
Ah, got it. Will make the change